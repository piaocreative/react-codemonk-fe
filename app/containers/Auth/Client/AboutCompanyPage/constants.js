export const key = 'aboutCompany';

export const CHANGE_NAME = 'CodeMonk/Client/AboutCompanyPage/CHANGE_NAME';
export const CHANGE_BRAND = 'CodeMonk/Client/AboutCompanyPage/CHANGE_BRAND';
export const CHANGE_REGISTER_NUMBER = 'CodeMonk/Client/AboutCompanyPage/CHANGE_REGISTER_NUMBER';
export const CHANGE_VAT_NUMBER = 'CodeMonk/Client/AboutCompanyPage/CHANGE_VAT_NUMBER';
export const CHANGE_INDUSTRY = 'CodeMonk/Client/AboutCompanyPage/CHANGE_INDUSTRY';
export const CHANGE_COMPANY_TYPE = 'CodeMonk/Client/AboutCompanyPage/CHANGE_COMPANY_TYPE';
export const COMPANY_CULTURES = 'CodeMonk/Client/AboutCompanyPage/COMPANY_CULTURES';

export const CHANGE_LINKEDIN_PROFILE = 'CHANGE_LINKEDIN_PROFILE';
export const CHANGE_GITHUB_PROFILE = 'CHANGE_GITHUB_PROFILE';
export const CHANGE_STACKOVERFLOW_PROFILE = 'CHANGE_STACKOVERFLOW_PROFILE';
export const CHANGE_DRIBBBLE_PROFILE = 'CHANGE_DRIBBBLE_PROFILE';
export const CHANGE_BEHANCE_PROFILE = 'CHANGE_BEHANCE_PROFILE';
export const CHANGE_PERSONAL_PROFILE = 'CHANGE_PERSONAL_PROFILE';

export const SUBMIT_ABOUT_COMPANY_FORM = 'CodeMonk/Client/AboutCompanyPage/SUBMIT_ABOUT_COMPANY_FORM';
