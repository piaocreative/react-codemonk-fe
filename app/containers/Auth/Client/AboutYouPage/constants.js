export const key = 'aboutYou';

export const CHANGE_FIRSTNAME = 'CodeMonk/Client/AboutYouPage/CHANGE_FIRSTNAME';
export const CHANGE_LASTNAME = 'CodeMonk/Client/AboutYouPage/CHANGE_LASTNAME';
export const CHANGE_COUNTRYCODE = 'CodeMonk/Client/AboutYouPage/CHANGE_COUNTRYCODE';
export const CHANGE_PHONENUMBER = 'CodeMonk/Client/AboutYouPage/CHANGE_PHONENUMBER';
export const CHANGE_JOB_TITLE = 'CodeMonk/Client/AboutYouPage/CHANGE_JOB_TITLE';
export const CHANGE_JOB_ROLE = 'CodeMonk/Client/AboutYouPage/CHANGE_JOB_ROLE';

export const SUBMIT_ABOUT_YOU_FORM = 'CodeMonk/Client/AboutYouPage/SUBMIT_ABOUT_YOU_FORM';

export const clientRoles = [
  {
    label: 'Recruitment Manager',
    value: 'recruitment-manager',
  },
  {
    label: 'Hiring Manager',
    value: 'hiring-manager',
  },
  {
    label: 'Billing & Admin',
    value: 'billing-admin',
  },
];
