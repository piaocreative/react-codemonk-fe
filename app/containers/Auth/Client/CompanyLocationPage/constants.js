export const key = 'companyLocation';

export const CHANGE_LOCATION_NAME = 'CodeMonk/Client/CompanyLocationPage/CHANGE_LOCATION_NAME';
export const CHANGE_POSTCODE = 'CodeMonk/Client/CompanyLocationPage/CHANGE_POSTCODE';
export const CHANGE_ADDRESSLINEONE = 'CodeMonk/Client/CompanyLocationPage/CHANGE_ADDRESSLINEONE';
export const CHANGE_ADDRESSLINETWO = 'CodeMonk/Client/CompanyLocationPage/CHANGE_ADDRESSLINETWO';
export const CHANGE_CITY = 'CodeMonk/Client/CompanyLocationPage/CHANGE_CITY';
export const CHANGE_COUNTRY = 'CodeMonk/Client/CompanyLocationPage/CHANGE_COUNTRY';
export const CHANGE_STATE = 'CodeMonk/Client/CompanyLocationPage/CHANGE_STATE';
export const CHANGE_TIMEZONE = 'CodeMonk/Client/CompanyLocationPage/CHANGE_TIMEZONE';

export const SUBMIT_COMPANY_LOCATION_FORM = 'CodeMonk/Client/AboutYouPage/SUBMIT_COMPANY_LOCATION_FORM';
