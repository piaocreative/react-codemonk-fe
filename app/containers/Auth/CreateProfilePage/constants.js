/*
 * CreateProfilePage Constants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const key = 'createProfile';

export const CHANGE_REGISTRATION_TYPE = 'CodeMonk/CreateProfilePage/CHANGE_REGISTRATION_TYPE';

export const CHANGE_FIRSTNAME = 'CodeMonk/CreateProfilePage/CHANGE_FIRSTNAME';
export const CHANGE_LASTNAME = 'CodeMonk/CreateProfilePage/CHANGE_LASTNAME';
export const CHANGE_COUNTRYCODE = 'CodeMonk/CreateProfilePage/CHANGE_COUNTRYCODE';
export const CHANGE_PHONENUMBER = 'CodeMonk/CreateProfilePage/CHANGE_PHONENUMBER';
export const CHANGE_POSTCODE = 'CodeMonk/CreateProfilePage/CHANGE_POSTCODE';
export const CHANGE_ADDRESSLINEONE = 'CodeMonk/CreateProfilePage/CHANGE_ADDRESSLINEONE';
export const CHANGE_ADDRESSLINETWO = 'CodeMonk/CreateProfilePage/CHANGE_ADDRESSLINETWO';
export const CHANGE_CITY = 'CodeMonk/CreateProfilePage/CHANGE_CITY';
export const CHANGE_COUNTRY = 'CodeMonk/CreateProfilePage/CHANGE_COUNTRY';
export const CHANGE_TIMEZONE = 'CodeMonk/CreateProfilePage/CHANGE_TIMEZONE';

export const CHANGE_COMPANY_DETAILS = 'CodeMonk/CreateProfilePage/CHANGE_COMPANY_DETAILS';

export const SAVE_FOR_LATER = 'CodeMonk/CreateProfilePage/SAVE_FOR_LATER';
export const SUBMIT_CREATE_PROFILE_FORM = 'CodeMonk/CreateProfilePage/SUBMIT_CREATE_PROFILE_FORM';

export const registrationTypes = [
  {
    id: 1,
    label: 'Individual',
    isChecked: false,
    value: 'individual',
    groupName: 'registrationType',
  },
  {
    id: 2,
    label: 'Company',
    isChecked: false,
    value: 'company',
    groupName: 'registrationType',
  },
];
