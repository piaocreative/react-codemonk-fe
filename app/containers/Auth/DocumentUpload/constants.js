export const key = 'documentUpload';

export const EXPERIENCE_SAVE_FOR_LATER = 'EXPERIENCE_SAVE_FOR_LATER';
export const ADD_EXPERIENCE = 'CodeMonk/WorkExperience/ADD_EXPERIENCE';

export const fieldNameArray = [
  'idProof',
  'addressProof',
  'companyIncorporationCertificateUrl',
  'companyVatRegistrationCertificateUrl',
  'companyInsuranceDocumentUrl',
];
