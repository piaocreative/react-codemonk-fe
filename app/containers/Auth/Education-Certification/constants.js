export const key = 'EducationDetailsForm';

const prefix = 'EDUCATION_DETAILS_FORM_';
export const CHANGE_EDUCATION_DETAILS = `${prefix}CHANGE_EDUCATION_DETAILS`;
export const CHANGE_CERTIFICATE_DETAILS = `${prefix}CHANGE_CERTIFICATE_DETAILS`;
export const SAVE_FOR_LATER = `${prefix}SAVE_FOR_LATER`;
export const EDIT_EDUCATION = `${prefix}EDIT_EDUCATION`;
export const EDIT_CERTIFICATE = `${prefix}EDIT_CERTIFICATE`;

export const ADD_EDUCATION = `CodeMonk/Education-Certification/ADD_EDUCATION`;
export const ADD_CERTIFICATE = `CodeMonk/Education-Certification/ADD_CERTIFICATE`;

export const CHANGE_COLLEGE = 'CodeMonk/Education-Certification/CHANGE_COLLEGE';
export const CHANGE_COUNTRY = 'CodeMonk/Education-Certification/CHANGE_COUNTRY';
export const CHANGE_START_YEAR = 'CodeMonk/Education-Certification/CHANGE_START_YEAR';
export const CHANGE_END_YEAR = 'CodeMonk/Education-Certification/CHANGE_END_YEAR';
export const CHANGE_DEGREE_TITLE = 'CodeMonk/Education-Certification/CHANGE_DEGREE_TITLE';
export const CHANGE_DEGREE_LEVEL = 'CodeMonk/Education-Certification/CHANGE_DEGREE_LEVEL';
export const CHANGE_CERTIFICATE_NAME = 'CodeMonk/Education-Certification/CHANGE_CERTIFICATE_NAME';
export const CHANGE_ORGANISATION = 'CodeMonk/Education-Certification/CHANGE_ORGANISATION';
export const CHANGE_DATE_OBTAINED = 'CodeMonk/Education-Certification/CHANGE_DATE_OBTAINED';
export const CHANGE_CERTIFICATE_URL = 'CodeMonk/Education-Certification/CHANGE_CERTIFICATE_URL';
