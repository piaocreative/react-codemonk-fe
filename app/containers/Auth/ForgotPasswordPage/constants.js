/*
 * ForgotPasswordPage Constants
 */
export const key = 'ForgotPasswordForm';
const prefix = 'CodeMonk/ForgotPassword/';
export const CHANGE_EMAIL = `${prefix}CHANGE_EMAIL`;
export const FORGOT_PASSWORD_SUBMIT = `${prefix}FORGOT_PASSWORD_SUBMIT`;
