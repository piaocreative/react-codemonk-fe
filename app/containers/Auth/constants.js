export const TALENT_STEP_1_URL = '/talent/about-you';
export const TALENT_STEP_2_URL = '/talent/qualifications';
export const TALENT_STEP_3_URL = '/talent/experience';
export const TALENT_STEP_4_URL = '/talent/projects';
export const TALENT_STEP_5_URL = '/talent/preferences';
export const TALENT_STEP_6_URL = '/talent/salary-billing';
export const TALENT_STEP_7_URL = '/talent/document-upload';

export const TALENT_STEP_1 = 'About you';
export const TALENT_STEP_2 = 'Qualifications';
export const TALENT_STEP_3 = 'Experience';
export const TALENT_STEP_4 = 'Projects';
export const TALENT_STEP_5 = 'Preferences';
export const TALENT_STEP_6 = 'Salary & Billing';
export const TALENT_STEP_7 = 'Document upload';

export const CLIENT_STEP_1 = 'About you';
export const CLIENT_STEP_2 = 'Company';
export const CLIENT_STEP_3 = 'Locations';

export const CLIENT_STEP_1_URL = '/client/about-you';
export const CLIENT_STEP_2_URL = '/client/about-company';
export const CLIENT_STEP_3_URL = '/client/company-location';

export const talentOnboardingSteps = [
  {
    step: 1,
    name: TALENT_STEP_1,
    url: TALENT_STEP_1_URL,
  },
  {
    step: 2,
    name: TALENT_STEP_2,
    url: TALENT_STEP_2_URL,
  },
  {
    step: 3,
    name: TALENT_STEP_3,
    url: TALENT_STEP_3_URL,
  },
  {
    step: 4,
    name: TALENT_STEP_4,
    url: TALENT_STEP_4_URL,
  },
  {
    step: 5,
    name: TALENT_STEP_5,
    url: TALENT_STEP_5_URL,
  },
  {
    step: 6,
    name: TALENT_STEP_6,
    url: TALENT_STEP_6_URL,
  },
  {
    step: 7,
    name: TALENT_STEP_7,
    url: TALENT_STEP_7_URL,
  },
];

export const agencyTalentOnboardingSteps = [
  { ...talentOnboardingSteps[0] },
  { ...talentOnboardingSteps[1] },
  { ...talentOnboardingSteps[2] },
  { ...talentOnboardingSteps[3] },
  { ...talentOnboardingSteps[4] },
];

export const clientOnboardingSteps = [
  {
    step: 1,
    name: CLIENT_STEP_1,
    url: CLIENT_STEP_1_URL,
  },
  {
    step: 2,
    name: CLIENT_STEP_2,
    url: CLIENT_STEP_2_URL,
  },
  {
    step: 3,
    name: CLIENT_STEP_3,
    url: CLIENT_STEP_3_URL,
  },
];
