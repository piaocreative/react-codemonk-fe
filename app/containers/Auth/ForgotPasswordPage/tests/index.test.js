import React from 'react';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import request from 'utils/request';
import { LOAD_REPOS } from '../../../App/constants';
import { ForgotPasswordPage as MainForm, mapDispatchToProp } from '../index';
import initialState from '../reducer';
import { CHANGE_EMAIL, FORGOT_PASSWORD_SUBMIT } from '../constants';
jest.mock('utils/request');
let store;
const mockStore = configureStore();
const props = {
  email: '',

  dispatch: jest.fn(),
  handleSubmit: jest.fn(),

  invalid: false,
  loading: false,
  responseSuccess: false,
  responseError: false,
  history: { push: jest.fn(), replace: jest.fn() },
  location: {},

  onChangeEmail: jest.fn(),
  onSubmitForm: jest.fn(),
};
const getWrapper = () => shallow(<MainForm store={store} {...props} />);
// const getInstance = () => getWrapper().instance();
const intializeSetup = () => {
  store = mockStore(initialState);
  request.mockImplementation(() => Promise.resolve({ status: 1 }));
};

describe('<MainForm />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<MainForm store={store} {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});

describe('mapDispatchToProp Testing', () => {
  intializeSetup();
  getWrapper();
  test('Testing if  CHANGE_EMAIL are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProp(dispatch).onChangeEmail({ target: { value: 'abcd@example.com' } });
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: 'abcd@example.com',
      type: CHANGE_EMAIL,
    });
  });

  test('Testing if the save for later LOAD_REPOS events are dispatched correctly', () => {
    const dispatch = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    mapDispatchToProp(dispatch).onSubmitForm(event);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: LOAD_REPOS,
    });
  });

  test('Testing if the save for later FORGOT_PASSWORD_SUBMIT events are dispatched correctly', () => {
    const dispatch = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    mapDispatchToProp(dispatch).onSubmitForm(event);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: FORGOT_PASSWORD_SUBMIT,
    });
  });
});
