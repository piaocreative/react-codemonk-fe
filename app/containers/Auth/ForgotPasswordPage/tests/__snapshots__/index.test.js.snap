// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainForm /> should render and match the snapshot 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Forgot password?
    </title>
    <meta
      content="Forgot password?"
      name="description"
    />
  </HelmetWrapper>
  <ForwardRef(ContainerMod)
    className="w-auto p-0"
  >
    <ForwardRef(Card)
      className="p-0 bg-transparent"
    >
      <form>
        <ForwardRef(OnboardingForm)>
          <ForwardRef(OnboardingForm__OnBoardingFormBody)>
            <React.Fragment>
              <ForwardRef(H1)
                className="mb-3"
              >
                <FormattedMessage
                  defaultMessage="Forgot password?"
                  id="CodeMonk.components.ForgotPasswordPage.headingForgotPassword"
                  values={Object {}}
                />
              </ForwardRef(H1)>
              <ForwardRef(P)
                className="p16 mb-5"
                opacityVal="0.5"
              >
                <FormattedMessage
                  defaultMessage="Enter your email to get a link to set up a new password"
                  id="CodeMonk.components.ForgotPasswordPage.textIntro"
                  values={Object {}}
                />
              </ForwardRef(P)>
            </React.Fragment>
            <FormGroup
              tag="div"
            >
              <ForwardRef(Label__FormLabel)>
                <FormattedMessage
                  defaultMessage="Email"
                  id="CodeMonk.containers.containerMessage.labelEmailAddress"
                  values={Object {}}
                />
              </ForwardRef(Label__FormLabel)>
              <ForwardRef
                component={[Function]}
                name="email"
                normalize={[Function]}
                onChange={[MockFunction]}
                placeholder="e.g. johndoe@gmail.com"
                type="text"
                validate={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
                value=""
              />
            </FormGroup>
            <div
              className="mt-4"
            >
              <Button
                className="btn-primary  w-100"
                disabled={false}
              >
                <FormattedMessage
                  defaultMessage="Send me a link"
                  id="CodeMonk.components.ForgotPasswordPage.buttonResetLink"
                  values={Object {}}
                />
              </Button>
              <ForwardRef(P)
                className="p16 mt-5"
              >
                <FormattedMessage
                  defaultMessage="Back to"
                  id="CodeMonk.components.ForgotPasswordPage.textBackTo"
                  values={Object {}}
                />
                <ForwardRef(A)
                  className="ms-1 text-decoration-underline"
                  href="/login"
                >
                  <FormattedMessage
                    defaultMessage="Login"
                    id="CodeMonk.components.ForgotPasswordPage.loginLink"
                    values={Object {}}
                  />
                </ForwardRef(A)>
              </ForwardRef(P)>
            </div>
          </ForwardRef(OnboardingForm__OnBoardingFormBody)>
        </ForwardRef(OnboardingForm)>
      </form>
    </ForwardRef(Card)>
  </ForwardRef(ContainerMod)>
</React.Fragment>
`;
